### PR TITLE
feat(backend): add V2 lightweight Python pipeline sample. Fixes #5689

### LINKED ISSUE #5689: [Samples] Add v2 compatible built-in samples

### PR DESCRIPTION
**Pull request by james-jwu**
Added data passing sample, mostly a copy of https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline.py

Added a parameter in sample_config.json to indicate compilation mode.

### CODE REVIEW

File: samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+from kfp import dsl
+from kfp import components
+from kfp.components import InputPath, OutputPath
+from kfp.v2.dsl import Input, Output, Dataset, Model, component
+import kfp.compiler as compiler
+
+
+@component
+def preprocess(
+    # An input parameter of type string.
```
Comment (NikeNano) [inline samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py:26]: nit: would it make sens to move these comments to the docstring instead? https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Comment (james-jwu) [reply samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py:26]: In principle I agree to using docstring consistently. However I feel the inlined comments here are more helpful to readers to demonstrate usage. I'd like to hear opinions from other reviewers as well.

Comment (chensun) [reply samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py:26]: Overall, I agree on using docstring for component interface description.
But for this specific sample, the comments here are sort of documentations that explain our new annotations: `Input[]`, `Output[]`, and how they compare to the existing annotations like `OutputPath`. 
In that sense, they are more like comments for technical details than docstrings which usually describe the meaning of the function arguments. So I agree with James that this seems to be more helpful to demonstrate usage. WDYT? @NikeNano

**Review by chensun — COMMENTED** [review]
/lgtm

File: samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py (hunk 1 above)
Comment (NikeNano) [reply samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py:26]: I prefer to have it in the doc string but I can see the advantage.

**Pull request merged**